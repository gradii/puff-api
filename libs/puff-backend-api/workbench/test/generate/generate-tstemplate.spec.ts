import * as ts from 'typescript';


describe('generate tstemplate', () => {
  it('test generate ast', () => {
    const ast = ts.createClassDeclaration(
      undefined,
      [ts.createModifier(ts.SyntaxKind.ExportKeyword)],
      ts.createIdentifier('StepStatus'),
      undefined,
      undefined,
      [
        ts.createProperty(
          undefined,
          [ts.createModifier(ts.SyntaxKind.StaticKeyword)],
          ts.createIdentifier('StartStatus'),
          undefined,
          undefined,
          ts.createStringLiteral('Running Step: StartStatus')
        ),
        ts.createProperty(
          undefined,
          [ts.createModifier(ts.SyntaxKind.StaticKeyword)],
          ts.createIdentifier('SuccessStatus'),
          undefined,
          undefined,
          ts.createStringLiteral('Running Step: SuccessStatus')
        ),
        ts.createProperty(
          undefined,
          [ts.createModifier(ts.SyntaxKind.StaticKeyword)],
          ts.createIdentifier('FailureStatus'),
          undefined,
          undefined,
          ts.createStringLiteral('Running Step: FailStatus')
        ),
        ts.createProperty(
          undefined,
          [ts.createModifier(ts.SyntaxKind.StaticKeyword)],
          ts.createIdentifier('FinishedStatus'),
          undefined,
          undefined,
          ts.createStringLiteral('Finished Step')
        )
      ]
    );

    const printer = ts.createPrinter({ newLine: ts.NewLineKind.LineFeed, omitTrailingSemicolon: true, removeComments: false });

    const resultFile = ts.createSourceFile('', '', ts.ScriptTarget.Latest,
      /*setParentNodes*/ false,
      ts.ScriptKind.TS);

    resultFile.statements = ts.createNodeArray([ast]);

    const result = printer.printNode(ts.EmitHint.Unspecified, resultFile, resultFile);

    expect(result).toMatchSnapshot()
  });
});
