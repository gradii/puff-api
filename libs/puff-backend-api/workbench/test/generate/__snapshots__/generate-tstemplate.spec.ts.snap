// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate tstemplate test generate ast 1`] = `
"export class StepStatus {
    static StartStatus = \\"Running Step: StartStatus\\";
    static SuccessStatus = \\"Running Step: SuccessStatus\\";
    static FailureStatus = \\"Running Step: FailStatus\\";
    static FinishedStatus = \\"Finished Step\\";
}
"
`;
